packages/apputils/src/sessioncontext.tsx

```diff
@@ -1297,13 +1348,26 @@ export const sessionContextDialogs: ISessionContext.IDialogs = {
       Dialog.okButton({ label: trans.__('Select') })
     ];
 
+    const selectPreferredKernel = this._settings?.get('selectPreferredKernel')
+      .composite as boolean;
+
     const dialog = new Dialog({
       title: trans.__('Select Kernel'),
       body: new Private.KernelSelector(sessionContext, translator),
-      buttons
+      buttons,
+      checkbox: {
+        label: trans.__('Always select the preferred kernel'),
+        caption: trans.__(
+          'Remember my choice and always select the preferred kernel'
+        ),
+        checked: selectPreferredKernel
+      }
     });
 
     const result = await dialog.launch();
+
+    this._settings?.set('selectPreferredKernel', result.isChecked);
```

Comment:
We need a `catch` here.

```suggestion
    this._settings?.set('selectPreferredKernel', result.isChecked).catch(reason => { console.error("Failed to set 'selectPreferredKernel';\n", reason) });
```